### DISCUSSION TITLE
Plugin APIs always return login page

### DISCUSSION THREAD

**Discussion by LWJ-SARA** [mattieserver/netbox-topology-views#554] [Q&A] [answered]
NetBox version
v4.0.9

Topology Views version
v4.0.0

Steps to Reproduce
Create an API token for the desired user
Create an API Request to the netbox-topology-views' XML Export endpoint found on the REST API endpoint page using either POSTMAN or any programming language of your choice
Include API token as a header (Authorization)
Include query params for the options available
Send the request
Expected Behavior
Returns an XML response of the created topology

Observed Behavior
Returns the netbox login page (HTML) as a response

**Comment by mattieserver** [comment]
Does it work in your web browser if you browse to the API URL?
For me, it works if i go to `/api/plugins/netbox_topology_views/xml-export/?show_unconnected=True&show_cables=True&site_id=1&role_id=11&role_id=1&role_id=7&role_id=10&role_id=18&role_id=3&role_id=4`

**Reply by LWJ-SARA** [reply]
I put the URL of the API into the browser and it works like this:
![2](https://github.com/user-attachments/assets/e831e7df-8d1f-4ea0-83d9-d9df822ec6f2)
But in the postman, it will automatically add a cookie csrftoken = SZVBqQ4s0qnMGE1nfwrt9kvNcB5TgCEX, when I send the request, will return netbox login page.

**Reply by mattieserver** [reply]
https://github.com/netbox-community/netbox-topology-views/pull/556 should fix this issue

**Reply by mattieserver** [reply, marked as answer]
If you upgrade to v4.0.1 the Api should work
